### PR TITLE
SONARNTEST-30 Import tests execution time from test execution result reports

### PR DESCRIPTION
**Pull request by dbolkensteyn**
This PR supercedes #20

### CODE REVIEW

File: src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioTestResultsFileParser.java [hunk 1]
```diff
@@ -84,7 +94,42 @@ private void handleCountersTag() {
...
+      try {
+        value = keepOnlyMilliseconds(value);
+        return dateFormat.parse(value);
+      } catch (ParseException e) {
+        throw xmlParserHelper.parseError("Expected an valid date and time instead of \"" + value + "\" for the attribute \"" + name + "\". " + e.getMessage());
+      }
+    }
+
+    private String keepOnlyMilliseconds(String value) {
+      StringBuffer sb = new StringBuffer();
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioTestResultsFileParser.java]: MAJOR Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1149)

File: src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioTestResultsFileParser.java [hunk 2]
```diff
@@ -84,7 +94,42 @@ private void handleCountersTag() {
...
+
+      Matcher matcher = millisecondsPattern.matcher(value);
+      while (matcher.find()) {
+        String milliseconds = matcher.group(2);
+        String trailingZeros = "";
+        for (int i = 0; i < 3 - milliseconds.length(); i++) {
+          trailingZeros += "0";
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioTestResultsFileParser.java]: MAJOR Use a StringBuilder instead. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1643)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by dbolkensteyn** [comment]
@sandermvanvliet I've made a few changes to your original PR - feel free to test this one

**Pull request merged**